Azure/communication-ui-library-android
# [Calling][Feature] user facing diagnostics

## Purpose

This PR introduces User Facing Diagnostics notifications on the UI SDK.
User Facing Diagnostics are enabled by default and will show 2 types of notifications for Call Diagnostics during a call:

- A Top Message Bar that will be persistent unless the diagnostic is no longer relevant or the user dismisses the notification pressing the X button on the right.
- A type of Toast notification that appears on the bottom of the screen with information that will disappear after 4 seconds 

![image](https://github.com/Azure/communication-ui-library-android/assets/79163512/f5833f68-1129-4ed7-b7bc-557aea823393)
![image](https://github.com/Azure/communication-ui-library-android/assets/79163512/45bc4d20-eca9-41be-8477-c0d02f27349f)
![image](https://github.com/Azure/communication-ui-library-android/assets/79163512/f3d32a61-9c92-4718-bde5-675e0e760bdd)
![image](https://github.com/Azure/communication-ui-library-android/assets/79163512/f1ce6725-9840-4acc-ad7e-6a738d74d7b1)

## Does this introduce a breaking change?



## Pull Request Type
What kind of change does this Pull Request introduce?



## Pull Request Checklist




## How to Test
* Join a ACS group call or a Teams meeting.
* Once the call is connected, mute the microphone.
* Start speaking.
* See a notification on the bottom of the screen that says "You are muted"

* Open...
* Click on...


## What to Check
Verify that the following are valid
* ...

## Other Information
